functions/functionsv2/firebase/hellofirestore/hellofirestore_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hellofirestore
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/googleapis/google-cloudevents-go/cloud/firestoredata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func TestHelloFirestore(t *testing.T) {
+	ctx := context.Background()
+
+	// create a Firestore document event
+	data := &firestoredata.DocumentEventData{
+		OldValue: &firestoredata.Document{
+			Name: "oldDocumentName",
+		},
+		Value: &firestoredata.Document{
+			Name: "newDocumentName",
+		},
+	}
+	dataAny, err := anypb.New(data)
+	if err != nil {
+		t.Fatalf("anypb.New: %v", err)
+	}
+	dataBytes, err := proto.Marshal(dataAny)
+	if err != nil {
+		t.Fatalf("proto.Marshal: %v", err)
+	}
+
+	// create a CloudEvent with the Firestore document event data
+	ce := event.New("1.0")
+	ce.SetSource("test-source")
+	ce.SetType("google.cloud.firestore.document.v1.created")
+	ce.SetDataContentType("application/protobuf")
+	ce.SetData(ce.DataContentType(), dataBytes)
+
+	// capture the output of fmt.Println
+	oldStdout := os.Stdout
```

Comment:
question: We typically pass an `io.Writer` as an argument to the sample function. I suspect that this is not possible given that the function satisfy the argument type for `functions.CloudEvent()`. Is that correct?